gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/CheckUnusedDependenciesTask.java

```diff
@@ -190,6 +191,11 @@ public final void dependenciesConfiguration(Configuration dependenciesConfigurat
     }
 
     public final void sourceOnlyConfiguration(Configuration configuration) {
+        Preconditions.checkNotNull(configuration);
+        Preconditions.checkArgument(
+                configuration.isCanBeResolved(),
+                "May only add sourceOnlyConfiguration if it is resolvable: %s",
+                configuration);
```

Comment:
To be fair this method shouldn't exist, because I don't know in what situation users might actually use it.
The name is confusing now too since it implies source-only configurations need to be added, but we already add (ignore) all compileOnly configurations from all source sets.

But for now, let's at least throw eagerly if the user gave us a bad configuration.